baseplate/clients/thrift.py

```diff
@@ -166,89 +207,128 @@ def _call_thrift_method(self: Any, *args: Any, **kwargs: Any) -> Any:
+                        continue
+                    except (TApplicationException, TProtocolException):
+                        # these are subclasses of TException but aren't ones that
+                        # should be expected in the protocol. this is an error!
                         span.finish(exc_info=sys.exc_info())
+                        raise
+                    except Error as exc:
+                        # a 5xx error is an unexpected exception but not 5xx are
+                        # not.
+                        if 500 <= exc.code < 600:
+                            span.finish(exc_info=sys.exc_info())
+                        else:
+                            span.finish()
+                        raise
+                    except TException:
+                        # this is an expected exception, as defined in the IDL
+                        span.finish()
+                        raise
+                    except:  # noqa: E722
+                        # something unexpected happened
+                        span.finish(exc_info=sys.exc_info())
+                        raise
                     else:
+                        # a normal result
                         span.finish()
-                    raise
-                except TException:
-                    # this is an expected exception, as defined in the IDL
-                    span.finish()
-                    raise
-                except:  # noqa: E722
-                    # something unexpected happened
-                    span.finish(exc_info=sys.exc_info())
-                    raise
-                else:
-                    # a normal result
-                    span.finish()
-                    return result
-
-        raise TTransportException(
-            type=TTransportException.TIMED_OUT,
-            message=f"retry policy exhausted while attempting {self.namespace}.{name}, last error was: {last_error}",
-        )
+                        return result
+
+            raise TTransportException(
+                type=TTransportException.TIMED_OUT,
+                message=f"retry policy exhausted while attempting {self.namespace}.{name}, last error was: {last_error}",
+            )
+        finally:
+            thrift_success = "true"
+            exception_type = ""
+            baseplate_status = ""
+            baseplate_status_code = ""
+            exc_info = sys.exc_info()
+            if exc_info[0] is not None:
+                thrift_success = "false"
+                exception_type = exc_info[0].__name__
+                current_exc = exc_info[1]
+                if isinstance(current_exc, Error):
+                    baseplate_status_code = current_exc.code
+                    baseplate_status = ErrorCode()._VALUES_TO_NAMES.get(current_exc.code, "")
+
+            REQUEST_LATENCY.labels(
+                thrift_method=name,
+                thrift_client_name="",
```

Comment:
🔕 Assuming this is not set anywhere by clients/in the interface. For now, would the pool name be appropriate?

Reply:
👀  I forgot about this I need to do some more research

Reply:
It looks like originally Kris intended to use `self.namespace` back in 49f14a2bb24098024713d019994b035ee4e2bf7b